dlt/extract/extractors.py

```diff
@@ -249,7 +282,11 @@ def _apply_contract_filters(
                 if filtered_columns.get(rename_mapping[name]) is not None
             ]
             if removed_columns:
+                name = rename_mapping[removed_columns[0]]
+                mode = filtered_columns.get(name)
                 item = pyarrow.remove_columns(item, removed_columns)
+                # TODO: should we maybe emit dictionaries here? or are arrow tables better? not sure...
```

Comment:
should we send the arrow table here or emit dicts?

Reply:
Is it possible to freeze them before sending?